core/src/main/scala/com/salesforce/op/utils/stages/NameDetectUtils.scala

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2017, Salesforce.com, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.utils.stages
+
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.ser.std.StdSerializer
+import com.salesforce.op.features.types.NameStats.GenderStrings._
+import com.salesforce.op.features.types.Text
+import com.salesforce.op.stages.impl.feature.{GenderDetectStrategy, TextTokenizer}
+import com.salesforce.op.utils.json.{JsonLike, JsonUtils, SerDes}
+import com.twitter.algebird._
+import com.twitter.algebird.macros.caseclass
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.internal.Logging
+import org.apache.spark.ml.param.{BooleanParam, DoubleParam, ParamValidators, Params}
+import org.apache.spark.sql.{Encoder, Encoders, SparkSession}
+
+import scala.io.Source
+import scala.util.Try
+
+/**
+ * Provides shared helper functions and variables for name identification and name to gender transformation.
+ */
+private[op] trait NameDetectFun extends Logging with NameDetectParams {
+  import GenderDetectStrategy._
+  import NameDetectUtils._
+
+  private[op] implicit def textValue2String[T <: Text](input: T#Value): String = input.getOrElse("")
```

Comment:
do these function really provide that much value - `textValue2String` & `string2Text`? if you would like to check if `Text` value is empty do:
```scala
import com.salesforce.op.feature.types._
Text("abc") match {
   case SomeValue(Some(s)) => // we got some string value
   case _ => // handle empty case
}
```
or if you would like to check the string value length as well do:
```scala
Text("abc") match {
   case SomeValue(Some(s)) if s.length > 0 => // we got some non empty string value
   case _ => // handle empty case 
}
```